src/hotspot/share/memory/heapInspection.cpp

```diff
@@ -213,6 +213,8 @@ KlassInfoEntry* KlassInfoTable::lookup(Klass* k) {
 // Return false if the entry could not be recorded on account
 // of running out of space required to create a new entry.
 bool KlassInfoTable::record_instance(const oop obj) {
+  assert(!obj->is_forwarded(), "must not be forwarded");
```

Comment:
It'd be good if someone more familiar with this method/class can comment whether this assert makes sense in this context.